### ISSUE TITLE
Provide IREE host tools for Mac OS for cross-compilation

### ISSUE THREAD

**Issue by RechieKho** [iree-org/iree#16146]
### Request description

I would love to have IREE host tools for Mac OS included into the CI/CD. Currently, only IREE host tools for Linux is available. It is definitely a life saver for cross-compiling to android and embedded device. It would be nice to have the same luxury for cross-compiling to IOS, and Mac OS of different CPU architecture. It would be much time and energy saving if there is a readily available IREE host tools online to be downloaded instead of compiling the whole LLVM on our own.

### What component(s) does this issue relate to?

Other

**Comment by stellaraccident** [comment]
Good point. We should figure out how to distribute these. Macos capacity is always a challenge and I need to see where we are on it...
